### DISCUSSION TITLE
API for 3rd-party tool integrations

### DISCUSSION THREAD

**Discussion by ericscheid** [naturalcrit/homebrewery#1538] [Ideas]
Pinning a stub here for collecting nascent thoughts about facilitating 3rd-party tool integration.

See #1414, #1204, #1097, #809, #1532, #194, #162

- tools creating new brews (e.g. their tool confabulates a bunch of markdown for info they manage, they shunt that to HB as a brew)
- RSS feeds for watching tags / search terms
- editor-collaboration tools (since HB currently does a poor job of multi-user editing)

----

Examples of 3rd party tools that would be nice to build a pathway for..

- [Eigengrau's Essential Establishment Generator](https://eigengrausgenerator.com/)
- [Tetra-cube's Monster Generator](http://tetra-cube.com/dnd/dnd-statblock.html)
- [Alex Schroeder's Hex_Describe](https://alexschroeder.ch/wiki/Hex_Describe)
- [Watabou's One Page Dungeon generator](https://watabou.itch.io/one-page-dungeon) 
- [Fantasia Archive](https://www.reddit.com/r/worldbuilding/comments/o8eu4k/im_developing_a_100_free_local_oss_and_crossos/) #1414
-  [Monster Builder](http://rolladvantage.com/) - not yet public, see #194

The first couple have already taken a step or two on that pathway (i.e. they generate the MD for either THB or GMB to copy/paste)

**Comment by G-Ambatte** [comment]
There is some discussion in older PRs where Scott talked about outside apps being able to push a JSON object to `/api` in order to create a new Brew in the HB database, however based on previous discussions with @calculuschild, my current understanding is that direct creation of brews in the database should be discouraged and/or disallowed, with external apps instead being able to create a new brew in an Editor window, where the user can save it to their account, print, or discard it at their discretion.  
This was the thinking behind the `/template` Route (#1532) I created; the user is dropped into a NewPage editing environment, pre-populated with the parameters from the query. An issue with the implementation in that PR is that editing the data at `/template` will overwrite anything that the user has in `localStorage` from `/new` - user data may be lost in a way that is not obvious or intuitive.

**Comment by ericscheid** [comment]
Another possible 3rd party candidate is [Watabou's One Page Dungeon generator](https://watabou.itch.io/one-page-dungeon) (and village generator, and city generator, and hexmap generator).

Another is Alex Schroeder's generators (e.g. [Hex_Describe](https://alexschroeder.ch/wiki/Hex_Describe) et al).

**Reply by Gazook89** [reply]
More integrations...
- [Tetra-cube's Monster Generator](http://tetra-cube.com/dnd/dnd-statblock.html)
- [Table Generator](http://www.tablesgenerator.com)
- https://dungeonmapdoodler.com/draw/#

- remove.bg or similar


Just spitballin

**Reply by ericscheid** [reply]
[Tetra-cube's Monster Generator](http://tetra-cube.com/dnd/dnd-statblock.html) already has a "View Markdown" function (using legacy brew syntax) .. so that's an easy win (right now, their users would click that button and then wonder "what's a Hombrewery, where do I find that?")

Another interesting one is [Eigengrau's Essential Establishment Generator](https://eigengrausgenerator.com/) which also already has an export function for THB, GMB, Foundry, and Raw JSON.

Both require the user to copy, navigate to THB, open a new brew, paste, save, edit etc.

**Comment by 5e-Cleric** [comment]
#3481 adds an api endpoint will close #2638 for integrating with https://www.compassapp.info/, and other websites like that
